common/djangoapps/rocket_chat/views.py

```diff
@@ -0,0 +1,93 @@
+from django.http import Http404
+from django.contrib.auth.decorators import login_required
+
+from edxmako.shortcuts import render_to_response
+from opaque_keys.edx.keys import CourseKey
+
+from courseware.access import has_access
+from courseware.courses import get_course_with_access
+from student.models import CourseEnrollment
+
+from xmodule.modulestore.django import modulestore
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+from rocketchat_API.rocketchat import RocketChat as ApiRocketChat
+
+from .utils import create_course_group, create_user, get_rocket_chat_settings
+
+
+@login_required
+def rocket_chat_discussion(request, course_id):
+
+    if not configuration_helpers.get_value('ENABLE_ROCKET_CHAT_SERVICE'):
+        raise Http404
+
+    course_key = CourseKey.from_string(course_id)
+
+    with modulestore().bulk_operations(course_key):
+
+        user = request.user
+        course = get_course_with_access(user, 'load', course_key)
+
+        staff_access = has_access(user, 'staff', course)
+        user_is_enrolled = CourseEnrollment.is_enrolled(user, course.id)
+
+        course_homepage_invert_title = configuration_helpers.get_value(
+            'COURSE_HOMEPAGE_INVERT_TITLE', False)
+
+        course_title = course.display_name_with_default
+        if course_homepage_invert_title:
+            course_title = course.display_number_with_default
+
+        context = {
+            'request': request,
+            'cache': None,
+            'course': course,
+            'course_title': course_title,
+            'staff_access': staff_access,
+            'user_is_enrolled': user_is_enrolled,
+        }
+
+        rocket_chat_settings = get_rocket_chat_settings()
```

Comment:
What if this method returns none (because the configuration is not correct)? or even if the keys "admin_user", "admin_pass" or "public_url_service" are not defined? should we render a template saying there was an error and log something? or raise an ImproperlyConfigured exception (https://docs.djangoproject.com/en/1.8/ref/exceptions/#improperlyconfigured)?

Reply:
What do you prefer, rendering an alternative html saying that rocket chat is not available and also logging something? or raising the exception?

Reply:
Currently if the settings are None the view will show a message (line 91)  I am going to add somo logs to track the possible problems

Reply:
Yap you're right, what an idiot, the not covered case is when the settings are not well defined (some of the required keys is not present)

Reply:
ok